### PR TITLE
[inductor] Add fallback for collectives size estimation for unbacked

### PR DESCRIPTION
**Pull request by IvanKobzarev**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #127562

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

Differential Revision: [D57982928](https://our.internmc.facebook.com/intern/diff/D57982928)

### CODE REVIEW

**Comment by IvanKobzarev** [comment]
@IvanKobzarev has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D57982928).

**Review by yifuwang — APPROVED** [review]
LG!

File: torch/_inductor/comm_analysis.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def get_collective_input_size_bytes(node: ir.IRNode) -> int:
             # For ease of testing
             numel = int(numel)
         else:
-            numel = V.graph.sizevars.size_hint(numel)
+            numel = V.graph.sizevars.size_hint(numel, fallback=0)
```
Comment (ezyang) [inline torch/_inductor/comm_analysis.py:62]: Zero seems like a really bad fallback size to me lol.

**Pull request closed without merging**